app/models/manager_refresh/inventory/persister.rb

```diff
@@ -0,0 +1,95 @@
+class ManagerRefresh::Inventory::Persister
+  attr_reader :manager, :target, :collections
+
+  # @param manager [ManageIQ::Providers::BaseManager] A manager object
+  # @param target [Object] A refresh Target object
+  def initialize(manager, refresh_target)
+    @manager = manager
+    @target  = refresh_target
+
+    @collections = {}
+
+    initialize_inventory_collections
+  end
+
+  def options
+    @options ||= Settings.ems_refresh[manager.class.ems_type]
+  end
+
+  def inventory_collections
+    collections.values
+  end
+
+  def inventory_collections_names
+    collections.keys
+  end
+
+  def method_missing(method_name, *arguments, &block)
+    if inventory_collections_names.include?(method_name)
+      self.class.define_collections_reader(method_name)
+      send(method_name)
+    else
+      super
+    end
+  end
+
+  def respond_to_missing?(method_name, _include_private = false)
+    inventory_collections_names.include?(method_name) || super
+  end
+
+  def self.define_collections_reader(collection_key)
+    define_method(collection_key) do
+      collections.try(:[], collection_key)
+    end
+  end
+
+  protected
+
+  def initialize_inventory_collections
+    raise NotImplementedError, _("must be implemented in a subclass")
+  end
+
+  # Adds 1 ManagerRefresh::InventoryCollection under a target.collections using :association key as index
+  #
+  # @param inventory_collection_data [Hash] Hash used for ManagerRefresh::InventoryCollection initialize
+  def add_inventory_collection(inventory_collection_data)
+    data = inventory_collection_data
+    data[:parent] ||= manager
+
+    if !data.key?(:delete_method) && data[:model_class]
```

Comment:
What happens if there is no `:delete_method` and not `:model_class` ?

If thats a problem, I think that should be guarded in the initializer of `InventoryCollection`

Maybe not now.